gen/flights/ai_flight_planner.py

```diff
@@ -583,26 +583,14 @@ def critical_missions(self) -> Iterator[ProposedMission]:
 
         # Find friendly CPs within 100 nmi from an enemy airfield, plan CAP.
         for cp in self.objective_finder.vulnerable_control_points():
-            # Plan three rounds of CAP to give ~90 minutes coverage. Spacing
-            # these out appropriately is done in stagger_missions.
-            yield ProposedMission(
-                cp,
-                [
-                    ProposedFlight(FlightType.BARCAP, 2, self.MAX_CAP_RANGE),
-                ],
-            )
-            yield ProposedMission(
-                cp,
-                [
-                    ProposedFlight(FlightType.BARCAP, 2, self.MAX_CAP_RANGE),
-                ],
-            )
-            yield ProposedMission(
-                cp,
-                [
-                    ProposedFlight(FlightType.BARCAP, 2, self.MAX_CAP_RANGE),
-                ],
-            )
+            # Plan one round of CAP for each 30 minutes. If the last round is less then 15 minutes left, don't do it.
+            for minutes in range(1, self.game.settings.mission_length, 30):
```

Comment:
Rename `minutes` to just `_` to mark it as an unused variable.

`30` should be `cap_duration` from `Doctrine`. I think `self.faction.doctrin.cap_duration.total_seconds() // 60` is your way to get there from here. Or, since you don't actually care about the value in minutes and the second value should become a `timedelta` too, just deal with it in seconds. I think you'll end up with:

```
for _ in range(
    0,
    int(self.game.settings.desired_player_mission_duration.total_seconds()),
    int(self.faction.doctrine.cap_duration.total_seconds()),
):
```

(why 1 and not 0? afaict they're the same, but I saw you actually changed that and I'm curious what I'm missing)

Reply:
Fiddling with this a bit in the REPL to convince myself as well and it looks like the behavior difference between 0 and 1 here is that with 1 and a mission length of 91 minutes, three rounds are planned instead of four. It's pretty meaningless since it's just one minute of difference, but I suppose 0 is slightly more correct :)